src/user/service/user.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Repository, ObjectLiteral, DataSource } from 'typeorm';
+
+import BaseService from '../../common/service/base.service';
+import BadRequestException from '../../common/exception/badRequest.exception';
+import NotFoundException from '../../common/exception/notFound.exception';
+import UnauthorizedException from '../../common/exception/unauthorized.exception';
+
+import User from '../../auth/entity/user.entity';
+
+import Preference from '../entity/preference.entity';
+import UserPreferenceDTO from '../dto/user.preference.dto';
+import { UpsertUserPreferenceResponseLocals } from '../response/upsertUserPreference.response';
+
+class UserService implements BaseService {
+    userRepository: Repository<ObjectLiteral>;
+    preferenceRepository: Repository<ObjectLiteral>;
+
+    constructor(database: DataSource) {
+        this.userRepository = database.getRepository(User);
+        this.preferenceRepository = database.getRepository(Preference);
+    }
+
+    public upsertUserPreference = async (body: UserPreferenceDTO, locals: UpsertUserPreferenceResponseLocals) => {
+        try {
+            const preferenceID = body.preferenceID;
+            const preferences = await this.getPreferences(preferenceID);
+            if (!preferences) throw new BadRequestException('Preferences Invalid');
+
+            const email = locals.email;
+            const username = locals.username;
+            const user = await this.getUserByEmailAndUsername(email, username);
+            if (!user) throw new NotFoundException('User does not exist.');
+            if (!user.isVerified)
```

Comment:
This check can be moved to auth middleware and probably store whether the user is verified or not on jwt

Reply:
Do kindly check the latest commit 🙂